frame/dapps-staking/src/testing_utils.rs

```diff
@@ -15,19 +15,32 @@ pub(crate) struct MemorySnapshot {
 
 impl MemorySnapshot {
     /// Prepares a new `MemorySnapshot` struct based on the given arguments.
+    /// If the account delegated the rewards to another account, the free balance will be that of the account
     pub(crate) fn all(
         era: EraIndex,
         contract_id: &MockSmartContract<AccountId>,
         account: AccountId,
     ) -> Self {
-        Self {
-            era_info: DappsStaking::general_era_info(era).unwrap(),
-            dapp_info: RegisteredDapps::<TestRuntime>::get(contract_id).unwrap(),
-            staker_info: GeneralStakerInfo::<TestRuntime>::get(&account, contract_id),
-            contract_info: DappsStaking::contract_stake_info(contract_id, era).unwrap_or_default(),
-            ledger: DappsStaking::ledger(&account),
-            free_balance: <TestRuntime as Config>::Currency::free_balance(&account),
+        if let Some(delegated_account) = DelegatedAccounts::<TestRuntime>::get(account, contract_id) {
+           return Self {
+                era_info: DappsStaking::general_era_info(era).unwrap(),
+                dapp_info: RegisteredDapps::<TestRuntime>::get(contract_id).unwrap(),
+                staker_info: GeneralStakerInfo::<TestRuntime>::get(&account, contract_id),
+                contract_info: DappsStaking::contract_stake_info(contract_id, era).unwrap_or_default(),
+                ledger: DappsStaking::ledger(&account),
+                free_balance: <TestRuntime as Config>::Currency::free_balance(&delegated_account),
```

Comment:
This is hacky, and you loose the opportunity to verify that free balance of staker hasn't been changed in case of delegated reward.

Also, lots of code duplication that can be avoided.

Reply:
I agree on that missing case. I left the MemorySnapshot as it was but changed the assert_restake_reward function in order to assert correctly when the rewards were delegated.